src/controllers/projectController.js

```diff
@@ -0,0 +1,151 @@
+const saveFigmaDataAsNestedStructure = require("../utils/saveFigmaDataAsNestedStructure");
+
+exports.getAllVersions = async function (req, res) {
+  const { projectId } = req.params;
+  const accessToken = req.headers.authorization;
+
+  try {
+    const getVersions = await fetch(
+      `https://api.figma.com/v1/files/${projectId}/versions`,
+      {
+        method: "GET",
+        headers: {
+          "X-FIGMA-TOKEN": accessToken,
+        },
+      },
+    );
+
+    const responseJson = await getVersions.json();
+    const { status, versions } = responseJson;
+
+    if (status === 400) {
+      return res.status(400).send({
+        result: "error",
+        error: {
+          message: "잘못된 사용자 요청입니다. Figma Token을 확인해주세요.",
+        },
+      });
+    }
+
+    if (status === 403) {
+      return res.status(403).send({
+        result: "error",
+        error: {
+          message: "인증되지 않은 사용자입니다.",
+        },
+      });
+    }
+
+    if (status === 404) {
+      return res.status(404).send({
+        result: "error",
+        error: {
+          message: "존재하지 않는 피그마 파일입니다. 다시 입력해주세요.",
+        },
+      });
+    }
+
+    if (status === 500) {
+      return res.status(500).send({
+        result: "error",
+        error: {
+          message: "프로젝트 용량이 너무 커 응답에 실패했습니다.",
+        },
+      });
+    }
+
+    if (versions.length < 2) {
+      return res.status(502).send({
+        result: "error",
+        error: {
+          message: "비교할 수 있는 버전이 없어서 서비스 이용이 불가합니다.",
+        },
+      });
+    }
+
+    return res.status(200).json(responseJson);
+  } catch (err) {
+    return res.status(500).send({
+      result: "error",
+      error: {
+        message: "피그마 API로부터 응답이 없습니다.",
+      },
+    });
+  }
+};
+
+exports.getCommonPages = async function (req, res) {
+  const { projectId } = req.params;
+  const accessToken = req.headers.authorization;
+  const beforeVersion = req.query["before-version"];
+  const afterVersion = req.query["after-version"];
+
+  async function fetchFigmaData(projectKey, versionId) {
+    const figmaUrl = `https://api.figma.com/v1/files/${projectKey}?version=${versionId}`;
+
+    const responseJson = await fetch(figmaUrl, {
+      method: "GET",
+      headers: {
+        "X-FIGMA-TOKEN": accessToken,
+      },
+    });
+    const data = await responseJson.json();
+
+    return data;
+  }
+
+  async function getDocument(projectKey, versionId) {
+    const document = await Document.findOne({
+      projectKey,
+      versionId,
+    });
+
+    return document;
+  }
+
+  async function createDocument(projectKey, versionId) {
+    const figmaData = await fetchFigmaData(projectKey, versionId);
+    let document = await saveFigmaDataAsNestedStructure(
+      figmaData.document,
+      null,
+      0,
+    );
+
+    document.projectKey = projectKey;
+    document.versionId = versionId;
+
+    document = await Document.create(document);
+
+    return document;
+  }
```

Comment:
- `P3`
처음 `document`를 선언 할 때는 figma project의 tree구조를 받고있고
`await Document.create(document)`는 직렬화 된 트리를 return하고있는 것 같습니다!

각각 return하는 값의 의미가 다르고, 성질이 다른데 `document`라는 한개의 변수에 값을 재할당 하기 보다는
각각 `const`로 선언해서
```suggestion
    const document = await saveFigmaDataAsNestedStructure(
      figmaData.document,
      null,
      0,
    );

    document.projectKey = projectKey;
    document.versionId = versionId;

    const flattenedDocument = await Document.create(document);

    return flattenedDocument;
  }
```

이런식으로 작성해도 좋을 것 같습니다!
저는 `let`은 최대한 사용을 지양하는것이 좋다고 생각하기 때문입니다!

Reply:
저 또한 `let`를 최대한 지양하고 싶었는데 
제안해주신 방식이 더 좋은 것 같습니다! 

코드 가독성도 높이고 좋네요 👍